BaaS.md

```diff
@@ -0,0 +1,67 @@
+# Backup as a Service
+
+The Backup process entails a collection of ansible scripts that manipulate the servers to leave them in a “blank” state suitable for performing a consistent backup, a specific tool that is able to backup a complete Linux machine into an ISO image that later can be used to restore and an openstack CLI integration that allows the operator to perform the actions in a very easy way.
+
+This backup automation has a lot of things built in and it is very helpful, but it still lacks some degree of integration. To solve that, we are going to implement what we call backup as a service. This service will be a piece of software that will run on a container in the openstack’s overcloud, exposing an API that can be accessed by third parties to trigger or program backups, to list all the history of backups that are stored in the system and allow to trigger restorations or delete old backups. It will even be capable of accepting backup programming and programmed cleanups.
```

Comment:
There are a lot of doubts that come to my mind.
- As a starting point I think this piece of software is kind of a wrapper of ReaR. (In the future it might be a different "engine"), We need to ensure to meet the RBAC requirements.
- Is this service going to run as manager/agent where the manager is going to manage how the backup is going to be executed?
- is this service going to be valid for the undercloud. The RFE only comments the overcloud?
- Not sure about the kind of architecture where the agent-nodes allow their manager to delete their backups.

Reply:
1.- No problem, where can I see the RBAC requirements and what does this entitle?
2.- Not exactly, I was thinking on re-using the ansible role we are already working on. So, it would be a "manager" that runs on the undercloud (the text is wrong in this case) and runs ansible against the nodes that need to be backuped.
3.- Yes, in its first incarnation, it will be an API for the ansible role, so both the undercloud and the overcloud can be backuped, no matter where the service is deployed.
4. The delete action will be included in the ansible playbook, it will just connect to the node and delete the needed backups. No "agent-nodes" will exist.

Reply:
Not sure were to start about the RBAC. But there are some reference to start searching:
 - http://lists.openstack.org/pipermail/openstack-discuss/2021-April/022117.html
 - https://etherpad.opendev.org/p/policy-popup-xena-ptg

I need to think a little bit deeper before talking about the point 3.